### PR TITLE
pkg/handlers: implement search

### PR DESCRIPTION
**Pull request by aleh-null**
A very simple case insensitive search. Implementation piggybacks on `ListFiles()` functionality, that holds file content and some meta information like line number. 

TODO: 
- [ ] test
- [ ] concurrency
- [ ] grep
- [ ] match case
- [ ] match whole word

### CODE REVIEW

File: pkg/handlers/search_files.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+const (
+	queryKey = "query"
+)
+
+type SearchFilesHandler struct {
+	ProjectManager project.Manager
+}
+
+func (h SearchFilesHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```
Comment (artmoskvin) [inline pkg/handlers/search_files.go:23]: Are we using `type` and `content` anywhere here?

File: pkg/handlers/search_files.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	projectID, err := getProjectID(r)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Invalid project ID: %s", err), http.StatusBadRequest)
+		return
+	}
+	log.Info().Msgf("project id %s", projectID)
```
Comment (artmoskvin) [inline pkg/handlers/search_files.go:29]: Probably worth to reduce the level to DEBUG and add a more meaningful message

**Comment by artmoskvin** [comment]
Great start! 🚀

File: pkg/handlers/search_files.go (hunk 2 above)
Comment (aleh-null) [reply pkg/handlers/search_files.go:29]: Forgot to delete this line.

File: pkg/handlers/search_files.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+		http.Error(w, fmt.Sprintf("Failed to search: %s", err), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(result)
+}
+
+func getChecker(typ, query string) (check func(s string) bool, err error) {
```
Comment (artmoskvin) [inline pkg/handlers/search_files.go:68]: What's the benefit of using named return result here?

File: pkg/handlers/search_files.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+	switch typ {
+	case "exact":
+		return exactSearch(query)
+	case "grep":
+		return grepSearch(query)
+	default:
+		return caseInsensitiveSearch(query)
+	}
+}
```
Comment (artmoskvin) [inline pkg/handlers/search_files.go:77]: I'd say these are more like search options, not search type. The search type is the same – content search.

Comment (aleh-null) [reply pkg/handlers/search_files.go:77]: In this case non, just symmetric to the signature of function called here.

File: pkg/handlers/search_files.go (hunk 3 above)
Comment (aleh-null) [reply pkg/handlers/search_files.go:68]: In this case non, just symmetric to the signature of function called here.

File: pkg/handlers/search_files.go (hunk 1 above)
Comment (aleh-null) [reply pkg/handlers/search_files.go:27]: Let's discuss this. Depends how we want to do with file (name) search.

**Review by artmoskvin — APPROVED** [review]

**Pull request merged**